src/code-gen.ts

```diff
@@ -0,0 +1,228 @@
+          ? ts.factory.createQualifiedName(
+            ts.factory.createIdentifier(ctx.prefixRef),
+            ts.factory.createIdentifier(typeName)
+          )
+          : ts.factory.createIdentifier(typeName)
+        return ts.factory.createTypeReferenceNode(
+          typeNameNode,
+          [ts.factory.createTypeReferenceNode(ts.factory.createIdentifier(ref))]
+        )
+      }
+      return ts.factory.createTypeReferenceNode(ts.factory.createIdentifier(ref))
+    }
+    if (schema.allOf) {
+      const types = schema.allOf.map(s => CodeGen.generateTypeNodeForSchema(s, ctx))
+      if (types.length < 2) {
+        return types[0]
+      }
+      return ts.factory.createIntersectionTypeNode(types)
+    }
+    if (schema.oneOf) {
+      const types = schema.oneOf.map(s => CodeGen.generateTypeNodeForSchema(s, ctx))
+      if (types.length < 2) {
+        return types[0]
+      }
+      return ts.factory.createUnionTypeNode(types)
+    }
+
+    if (schema.type === 'array') {
+      if (!schema.items) {
+        return ts.factory.createArrayTypeNode(
+          ts.factory.createKeywordTypeNode(ts.SyntaxKind.AnyKeyword)
+        )
+      }
+      return ts.factory.createArrayTypeNode(CodeGen.generateTypeNodeForSchema(schema.items, ctx))
+    }
+
+    if (schema.type === 'object') {
+      return CodeGen.generateTypeLiteral(schema, ctx)
+    }
+
+    if (schema.type === 'boolean') {
+      if (schema.enum) {
+        return ts.factory.createUnionTypeNode(
+          schema.enum.map(name =>
+            ts.factory.createTypeReferenceNode(ts.factory.createIdentifier(name))
+          )
+        )
+      }
+      return CodeGen.nullableNodeType(
+        ts.factory.createKeywordTypeNode(ts.SyntaxKind.BooleanKeyword),
+        !!schema.nullable
+      )
+    }
+
+    if (schema.type === 'integer' || schema.type === 'number') {
+      if (schema.enum) {
+        return ts.factory.createUnionTypeNode(
+          schema.enum.map(name =>
+            ts.factory.createLiteralTypeNode(ts.factory.createNumericLiteral(name))
+          )
+        )
+      }
+      return CodeGen.nullableNodeType(
+        ts.factory.createKeywordTypeNode(ts.SyntaxKind.NumberKeyword),
+        !!schema.nullable
+      )
+    }
+
+    if (schema.format === 'date' || schema.format === 'date-time') {
+      return CodeGen.nullableNodeType(
+        ts.factory.createTypeReferenceNode(ts.factory.createIdentifier('Date')),
+        !!schema.nullable
+      )
+    }
+
+    if (schema.type === 'string') {
+      if (schema.enum) {
+        const enumNodes = schema.enum.map(name =>
+          ts.factory.createLiteralTypeNode(ts.factory.createStringLiteral(name))
+        )
+        return enumNodes.length === 1
+          ? enumNodes[0]
+          : ts.factory.createUnionTypeNode(enumNodes)
+      }
+      return CodeGen.nullableNodeType(
+        ts.factory.createKeywordTypeNode(ts.SyntaxKind.StringKeyword),
+        !!schema.nullable
+      )
+    }
+
```

Comment:
I think here we miss the https://github.com/Eywek/oatyp/commit/bf0a7dbc11b6cc33908afece33190eec9c1a22a2 fix

Reply:
Added it! Nice catch! I must've missed it somehow